packages/fxa-payments-server/src/index.tsx

```diff
@@ -69,46 +68,95 @@ const parseParams = (params: string): ParsedParams => params
     return acc;
   }, {});
 
-// Parse params out of the location hash, then remove the hash.
-async function getHashParams() {
-  const hashParams = parseParams(window.location.hash);
-  window.history.replaceState('', document.title, window.location.pathname + window.location.search);
-  return hashParams;
-}
 
 const ACCESS_TOKEN_KEY = 'fxa-access-token';
-type getVerifiedAccessTokenArgs = { accessToken?: string | null };
+type getVerifiedAccessTokenArgs = {
+  code?: string | null,
+  state?: string | null,
+};
+
+function returnToSettings () {
+  // TODO: bounce through a login redirect to get back here with a token
+  console.log('returning to settings');
+  window.location.href = `${config.servers.content.url}/settings`;
+  return null;
+}
+
 async function getVerifiedAccessToken({
-  accessToken = ''
+  code = '',
+  state = ''
 }: getVerifiedAccessTokenArgs): Promise<string | null> {
-  if (accessToken) {
+  let accessToken;
+  /*if (accessToken) {
     localStorage.setItem(ACCESS_TOKEN_KEY, accessToken);
   } else {
     accessToken = localStorage.getItem(ACCESS_TOKEN_KEY);
+  }*/
+
+  let codeVerifier;
+  let cookieState;
+
+  try {
+    const pkceResult = await fetch(
+      `${config.servers.content.url}/payments-pkce`,
+      {
+        credentials: 'include',
+        headers: { 'Content-Type': 'application/json' },
+        method: 'get',
+      }
+    )
+    if (pkceResult.status !== 200) {
```

Comment:
If this is a standard `fetch`, do we also need to verify that `pkceResult.ok` is true?